llvm/lib/SYCLLowerIR/SpecConstants.cpp

```diff
@@ -946,8 +949,14 @@ PreservedAnalyses SpecConstantsPass::run(Module &M,
             updatePaddingInLastMDNode(Ctx, SCMetadata, Padding);
           }
 
-          SCMetadata[SymID] = generateSpecConstantMetadata(
-              M, SymID, SCTy, NextID, /* is native spec constant */ false);
+          if (sycl::utils::isSYCLNativeCPU(M) && isa<StructType>(DefaultValue->getType())) {
```

Comment:
I've changed this to always pass the default value type to `generateSpecConstantMetadata`, unfortunately we can't use the default type for everything because `replaceAllUsesWith` will fail since due to the ABI the type returned by the spec const builtin is different from the default type.